### PR TITLE
Fixing Void bug

### PR DESCRIPTION
**Pull request by 487959867345**
Pretty much as the title says fixes the bug were spectators take fall damage.

### CODE REVIEW

**Comment by SuperCoder7979** [comment]
Cool, thanks for the PR!

File: src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java [hunk 1]
```diff
@@ -152,13 +143,15 @@ public static void open(GameWorld gameWorld, BwMap map, BwConfig config, Multima
...
-
+            game.on(PlayerDamageListener.EVENT, active::onTakeDamage);
             game.on(ExplosionListener.EVENT, affectedBlocks -> {
                 affectedBlocks.removeIf(map::isProtectedBlock);
             });
         });
     }
 
+
+
```
Comment (haykam821) [inline src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java:154]: There should probably only be one line between methods here.

File: src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java [hunk 2]
```diff
@@ -211,7 +204,12 @@ private void addPlayer(ServerPlayerEntity player) {
             this.spawnLogic.spawnAtCenter(player);
         }
     }
-
+    private boolean onTakeDamage(ServerPlayerEntity player, DamageSource source, float v){
```
Comment (haykam821) [inline src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java:207]:
```suggestion
    private boolean onTakeDamage(ServerPlayerEntity player, DamageSource source, float amount) {
```

File: src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java [hunk 3]
```diff
@@ -211,7 +204,12 @@ private void addPlayer(ServerPlayerEntity player) {
...
+        if (source == DamageSource.OUT_OF_WORLD && player.isSpectator()){
```
Comment (haykam821) [inline src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java:208]:
```suggestion
        if (source == DamageSource.OUT_OF_WORLD && player.isSpectator()) {
```

File: src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java [hunk 4]
```diff
@@ -211,7 +204,12 @@ private void addPlayer(ServerPlayerEntity player) {
...
+            return true;
```
Comment (haykam821) [inline src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java:209]: This could be condensed into a single `return` line.

Comment (gegy1000) [reply src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java:209]: It'd also probably be valid to cancel all damage to spectators- not strictly necessary though. I think the only damage that can apply to spectators is void damage anyway.

**Comment by gegy1000** [comment]
Merged in https://github.com/NucleoidMC/bed-wars/commit/0c165ace0d77217e3eeb704a9917d81fa0cb1ec1. Thanks for the contribution!

**Pull request closed without merging**